### PR TITLE
setTestNow() to be able to parse a string

### LINKED ISSUE #109: setTestNow() to be able to parse a string
Follow ~~https://github.com/briannesbitt/Carbon/pull/818~~ https://github.com/briannesbitt/Carbon/pull/847

### PR DESCRIPTION
**Pull request by serima**
Resolve #109

### CODE REVIEW

File: src/Traits/TestingAidTrait.php [hunk 1]
```diff
@@ -35,19 +35,20 @@
...
      *
-     * @param \Cake\Chronos\ChronosInterface|null $testNow The instance to use for all future instances.
+     * @param \Cake\Chronos\ChronosInterface|string|null $testNow The instance to use for all future instances.
      * @return void
      */
-    public static function setTestNow(ChronosInterface $testNow = null)
+    public static function setTestNow($testNow = null)
     {
-        static::$testNow = $testNow;
+        static::$testNow = is_string($testNow) ? static::parse($testNow) : $testNow;
```
Comment (lorenzo) [inline src/Traits/TestingAidTrait.php]: What happens if the string cannot be parsed?

Comment (serima) [inline src/Traits/TestingAidTrait.php]: It throws `\Exception`. It depends on `ChronosInterface::parse()` method.

#### Examples

* Case 1
Run `$class::setTestNow('2016-xx-xx');` in the test case.
Got `Exception: DateTime::__construct(): Failed to parse time string (2016-xx-xx) at position 4 (-): Unexpected character`

* Case 2
Run `$class::setTestNow('xx');` in it.
Got `Exception: DateTime::__construct(): Failed to parse time string (xx) at position 0 (x): The timezone could not be found in the database`

Comment (lorenzo) [inline src/Traits/TestingAidTrait.php]: Perfect :)

**Comment by lorenzo** [comment]
thanks for doing this!

**Pull request merged**